src/transformers/trainer.py

```diff
@@ -154,6 +159,9 @@
     from fairscale.optim.grad_scaler import ShardedGradScaler
 
 
+if is_deepspeed_available():
+    import deepspeed
```

Comment:
@sgugger - Should we perhaps not import ds by default and postpone it until we get `args.deepspeed`?

Reply:
I moved all the other imports inside `_init_deepspeed`

Reply:
(related: should we do the same for fairscale? why load a bunch of libraries if we don't know if we are going to use them)

Reply:
The import will need to go when necessary only, yes. Same for fairscale, especially with #9441 merged.